ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/LogSegment.java

```diff
@@ -231,6 +235,7 @@ private File getSegmentFile() {
   private volatile long endIndex;
   private RaftStorage storage;
   private RaftLogMetrics raftLogMetrics;
+  private final boolean stateMachineCachingEnabled;
```

Comment:
Let's pass the information in the method calls instead of add stateMachineCachingEnabled to LogSegment.